content/installation/upgrading-manager.md

```diff
@@ -2,96 +2,113 @@
 
-{{% gsNote title="Caution" %}}
-Make sure that no users are connected to any manager services during the upgrade or migration.
+When you upgrade to Cloudify to 4.2 and above, you must also consider [user roles and permissions]({{< relref "installation/upgrading-manager.md#upgrading-into-the-new-roles-system" >}}).
 {{% /gsNote %}}
 
-The upgrade or migration process includes:
-
-* Reinstall -
-
-  1. Save snapshot of the Cloudify Manager.
-  1. Uninstall the Cloudify Manager from the host
-  1. Install the Cloudify Manager on the host.
-  1. Restore snapshot of the Cloudify Manager to the host.
-
-* Migration -
+{{% gsWarning title="Downtime" %}}
+Make sure that no users are connected to any manager services during the upgrade or migration.
+{{% /gsWarning %}}
 
-  1. Save snapshot of the Cloudify Manager.
-  1. Install the Cloudify Manager on the host.
-  1. Restore snapshot of the Cloudify Manager to the host.
-  1. (Optional) Uninstall (teardown in Cloudify 4.2 and below) the Cloudify Manager from the host.
-  1. Migrate agents from the old Cloudify Manager.
+The upgrade process includes:
 
 * In-place upgrade -
 
-  1. Save snapshot of Cloudify Manager.
-  1. Uninstall (teardown in Cloudify 4.2 and below) the Cloudify Manager from the host.
-  1. Install the Cloudify Manager latest version on the host.
-  1. Restore snapshot of the Cloudify Manager to the host.
+  1. [Backup a snapshot]({{< relref "installation/upgrading-manager.md#backup-a-snapshot" >}}) of the Cloudify Manager.
+  1. [Backup the agents certificates]({{< relref "installation/upgrading-manager.md#backup-agent-certificates" >}}) from the Cloudify Manager.
+  1. [Uninstall the Cloudify Manager]({{< relref "installation/upgrading-manager.md#uninstall-cloudify-manager" >}}) from the original host. In Cloudify 4.2 and below, this is called teardown.
+  1. [Install the Cloudify Manager]({{< relref "installation/upgrading-manager.md#install-cloudify-manager" >}}) latest version on the host.
+  1. [Restore the snapshot]({{< relref "installation/upgrading-manager.md#restore-snapshot" >}}) of the Cloudify Manager to the host.
+  1. [Migrate agents]({{< relref "installation/upgrading-manager.md#migrate-agents" >}}) to the new Cloudify Manager.
 
 * Migration upgrade -
 
-  1. Save snapshot of the the Cloudify Manager.
-  1. Install the Cloudify Manager on the host.
-  1. Restore snapshot of the Cloudify Manager to the host.
-  1. (Optional) Uninstall (teardown in Cloudify 4.2 and below) the Cloudify Manager from the host.
-  1. Migrate agents from the old Cloudify Manager.
+  1. [Backup a snapshot]({{< relref "installation/upgrading-manager.md#backup-a-snapshot" >}}) of the Cloudify Manager.
+  1. [Backup the agents certificates]({{< relref "installation/upgrading-manager.md#backup-agent-certificates" >}}) from the Cloudify Manager.
```

Comment:
Not required here as far as I'm aware.

Reply:
(because the agents will still be connected to the old manager, and when you run cfy_agents install --all-tenants they'll be pointed to the new manager using the old manager to pass them tasks)

(also, the certificate will be wrong for the new manager unless it has the same IP address and/or hostname that the agents use to refer to the manager)